src/components/NewDatePicker/CalendarPicker/ScreenSlideAnimation.js

```diff
@@ -0,0 +1,74 @@
+import {Portal} from '@gorhom/portal';
+import PropTypes from 'prop-types';
+import React from 'react';
+import Animated, {runOnJS, useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+import CONST from '../../../CONST';
+import styles from '../../../styles/styles';
+import variables from '../../../styles/variables';
+import withWindowDimensions, {windowDimensionsPropTypes} from '../../withWindowDimensions';
+
+const propTypes = {
+    /** Children to render */
+    children: PropTypes.node.isRequired,
+
+    /** TestID for test */
+    testID: PropTypes.string,
+
+    ...windowDimensionsPropTypes,
+};
+
+const defaultProps = {
+    testID: undefined,
+};
+
+const ScreenSlideAnimation = React.forwardRef(({children, testID, isSmallScreenWidth, windowHeight}, ref) => {
+    const [visible, setVisible] = React.useState(false);
+    const width = isSmallScreenWidth ? windowHeight : variables.sideBarWidth;
+    const translateX = useSharedValue(width);
+
+    React.useImperativeHandle(
+        ref,
+        () => ({
+            close: () => {
+                translateX.value = withTiming(width, {duration: CONST.ANIMATED_TRANSITION}, (finished) => {
+                    if (!finished) {
+                        return;
+                    }
+                    runOnJS(setVisible)(false);
+                });
+            },
+            open: () => setVisible(true),
+        }),
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [width],
```

Comment:
What happens if you remove the `eslint-disable` and put `translateX` in the dependency array? does it cause an infinite loop?

I'm not too familiar with Reanimated, but thought `useSharedValue` returned a ref, so it could be safe to add in the deps array

Reply:
@thiagobrez It doesn't cause an infinite loop. I generally prefer not to include unnecessary variables in the dependency array. I like to keep a clear understanding of which variables an effect depends on.

Reply:
I'll seek out an ESLint configuration that can ignore such references.

Reply:
I understand, but if it causes no harm in adding it to the deps array, I would include it. Otherwise in the future, when someone is refactoring this, it can lead to bugs we are not aware of right now. Also, if it's a stable reference, it won't trigger a re-render anyway

Reply:
Sure, let's add that then